homeassistant/components/touchline/climate.py

```diff
@@ -36,38 +58,45 @@ class Touchline(ClimateDevice):
     """Representation of a Touchline device."""
 
     def __init__(self, touchline_thermostat):
-        """Initialize the climate device."""
+        """Initialize the Touchline device."""
         self.unit = touchline_thermostat
         self._name = None
         self._current_temperature = None
         self._target_temperature = None
+        self._current_operation_mode = None
+        self._current_week_program = None
+        self._current_preset_mode = None
+        self._preset_mode = None
+        self._preset_modes = PRESET_MODES
 
     @property
     def supported_features(self):
         """Return the list of supported features."""
         return SUPPORT_FLAGS
 
     def update(self):
-        """Update unit attributes."""
+        """Update thermostat attributes."""
         self.unit.update()
         self._name = self.unit.get_name()
         self._current_temperature = self.unit.get_current_temperature()
         self._target_temperature = self.unit.get_target_temperature()
+        self._current_operation_mode = self.unit.get_operation_mode()
```

Comment:
We don't need to store the operation mode in an instance attribute since it's only used locally on this method to calculate preset mode.

Reply:
Well, doing what you suggest, breaks the preset-mode, that isn't fetched/stored on HA restart. So HA doesn't poll the device. I'll just keep my version and (AGAIN!) close my PR. Too bad nobody else will profit from my changes, but I feel like I'm wasting my time here :)